src/utils/utils.ts

```diff
@@ -1346,25 +1347,44 @@ export const getTokenPrice = async (
   address: string,
   coinGeckoId?: string
 ): Promise<TokenPriceData> => {
-  const jupPrice = await getJupTokenPrice(address)
+  const cachedLastQueryTimestamp = localStorage.getItem('TOKEN_PRICE_LAST_QUERY_TIMESTAMP')
 
-  if (jupPrice.price !== 0) {
-    return jupPrice
-  } else if (coinGeckoId) {
-    const coingeckoPrice = await getCoinGeckoTokenPrice(coinGeckoId)
+  const lastQueryTimestamp = cachedLastQueryTimestamp ? Number(cachedLastQueryTimestamp) : 0
+  const cachedPriceData = localStorage.getItem(`TOKEN_PRICE_DATA`)
+  const priceData = cachedPriceData ? JSON.parse(cachedPriceData) : {}
 
-    return {
-      price: coingeckoPrice?.current_price || 0,
-      buyPrice: coingeckoPrice?.current_price || 0,
-      sellPrice: coingeckoPrice?.current_price || 0
-    }
-  } else {
-    return {
-      price: 0,
-      buyPrice: 0,
-      sellPrice: 0
+  let tokenPriceData = {
+    price: 0,
+    buyPrice: 0,
+    sellPrice: 0
+  }
+  if (!priceData[address] || Number(lastQueryTimestamp) + PRICE_QUERY_COOLDOWN <= Date.now()) {
+    try {
+      const jupPrice = await getJupTokenPrice(address)
+
+      if (jupPrice.price !== 0) {
+        tokenPriceData = jupPrice
+      } else if (coinGeckoId) {
```

Comment:
This could be improved to first check which tokens have `coingeckoId`, and if not then get jupTokenPrice. now there is a redundant call to sometimes